src/diffusers/utils/dynamic_modules_utils.py

```diff
@@ -37,6 +39,7 @@
 
 # See https://huggingface.co/datasets/diffusers/community-pipelines-mirror
 COMMUNITY_PIPELINES_MIRROR_ID = "diffusers/community-pipelines-mirror"
+_HF_REMOTE_CODE_LOCK = threading.Lock()
```

Comment:
Why do we have to acquire a lock?

Reply:
It's used in the step that modifies sys.modules. Assuming it is to avoid having background threads concurrently read/write from there during that operation. It's taken from `transformers` custom code loading.